### PR TITLE
[partners_api] cian api

### PR DESCRIPTION
**Pull request by milchakov**
https://jira.mail.ru/browse/MAPSME-4925

### CODE REVIEW

**Comment by milchakov** [comment]
@ygorshenin @rokuz PTAL

File: partners_api/cian_api.cpp [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    FromJSONObject(cluster, "lng", place.m_latlon.lon);
+
+    json_t * offers = json_object_get(cluster, "offers");
+
+    if (offers == nullptr)
+      return;
+
+    size_t const offersSize = json_array_size(offers);
+
+    if (offersSize == 0)
```
Comment (ygorshenin) [inline partners_api/cian_api.cpp]: Are you sure that we need to return here, not continue?

File: partners_api/cian_api.cpp [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  {
+    auto cluster = json_array_get(clusters, i);
+    cian::RentPlace place;
+    FromJSONObject(cluster, "lat", place.m_latlon.lat);
+    FromJSONObject(cluster, "lng", place.m_latlon.lon);
+
+    json_t * offers = json_object_get(cluster, "offers");
+
+    if (offers == nullptr)
+      return;
```
Comment (ygorshenin) [inline partners_api/cian_api.cpp]: Why not to continue here?

File: partners_api/cian_api.cpp [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+#include <utility>
+
+#include "3party/jansson/myjansson.hpp"
+
+using namespace platform;
+
+namespace
+{
+bool RunSimpleHttpRequest(string const & url, string & result)
```
Comment (ygorshenin) [inline partners_api/cian_api.cpp]: Looks like code from uber_api. Please, get rid of code-duplication.

Comment (ygorshenin) [inline partners_api/cian_api.cpp]: Probably, you'll be able to simplify code in booking api too.

File: partners_api/cian_api.cpp [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    try
+    {
+      MakeResult(rawResult, result);
+    }
+    catch (my::Json::Exception const & e)
+    {
+      LOG(LERROR, (e.Msg()));
+      result.clear();
+    }
+    GetPlatform().RunOnGuiThread([cb, result, reqId]() { cb(result, reqId); });
```
Comment (ygorshenin) [inline partners_api/cian_api.cpp]: Am I right that you don't want to expose possible errors to callback?

File: partners_api/cian_api.hpp [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+#include "base/worker_thread.hpp"
+
+#include <cstdint>
+#include <functional>
+#include <string>
+#include <vector>
+
+namespace cian
+{
+std::string const kBaseUrl = "https://api.cian.ru/rent-nearby/v1";
```
Comment (ygorshenin) [inline partners_api/cian_api.hpp]: Possible multiple definitions of the constant.

File: partners_api/cian_api.hpp [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  std::string m_address;
+};
+
+struct RentPlace
+{
+  ms::LatLon m_latlon;
+  std::vector<RentOffer> m_offers;
+};
+
+using RentNearbyCallback =
```
Comment (ygorshenin) [inline partners_api/cian_api.hpp]: Please, move this to the Api scope.

File: partners_api/cian_api.cpp (hunk 1 above)
Comment (milchakov) [inline partners_api/cian_api.cpp]: да, спасибо, заменил на continue

File: partners_api/cian_api.cpp (hunk 2 above)
Comment (milchakov) [inline partners_api/cian_api.cpp]: заменил на continue

File: partners_api/cian_api.cpp (hunk 3 above)
Comment (milchakov) [inline partners_api/cian_api.cpp]: сделал utils.hpp, вынес туда

File: partners_api/cian_api.cpp (hunk 4 above)
Comment (milchakov) [inline partners_api/cian_api.cpp]: да, возвращаю пустой список на случай ошибки

File: partners_api/cian_api.hpp (hunk 6 above)
Comment (milchakov) [inline partners_api/cian_api.hpp]: перенес

File: partners_api/cian_api.hpp (hunk 5 above)
Comment (milchakov) [inline partners_api/cian_api.hpp]: Не понял, она же в пространстве имен cian?

**Comment by milchakov** [comment]
@ygorshenin PTAL

File: partners_api/cian_api.hpp (hunk 5 above)
Comment (ygorshenin) [inline partners_api/cian_api.hpp]: So what? Imagine two individual cpp files, each includes cian_api.hpp. So, you have two definitions of the same constant, with the same name.

**Comment by milchakov** [comment]
@ygorshenin PTAL

File: partners_api/cian_api.cpp [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    MYTHROW(my::Json::Exception, ("The answer must contain a json object."));
+
+  json_t * clusters = json_object_get(root.get(), "clusters");
+
+  if (clusters == nullptr)
+    return;
+
+  size_t const clustersSize = json_array_size(clusters);
+
+  if (clustersSize == 0)
```
Comment (ygorshenin) [inline partners_api/cian_api.cpp]: No need in this check.

File: partners_api/utils.hpp [hunk 8]
```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include "platform/http_client.hpp"
```
Comment (ygorshenin) [inline partners_api/utils.hpp]: #include <string>

Comment (ygorshenin) [inline partners_api/utils.hpp]: ```#include <string>```

Comment (ygorshenin) [inline partners_api/utils.hpp]: And use ```std::string```.

File: partners_api/cian_api.cpp [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+
+namespace cian
+{
+std::string const kBaseUrl = "https://api.cian.ru/rent-nearby/v1";
+
+// static
+bool RawApi::GetRentNearby(m2::RectD const & rect, std::string & result,
+                           std::string const & baseUrl /* = kBaseUrl */)
+{
+  ostringstream url;
```
Comment (rokuz) [inline partners_api/cian_api.cpp]: std::

**Comment by milchakov** [comment]
@rokuz PTAL & Merge please

**Pull request merged**